app/assets/javascripts/school_absences/SchoolAbsencesDashboard.js

```diff
@@ -0,0 +1,404 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import moment from 'moment';
+import memoizer from '../helpers/memoizer';
+import {
+  supportsExcusedAbsences,
+  supportsHouse,
+  shouldDisplayHouse,
+  supportsCounselor,
+  shouldDisplayCounselor
+} from '../helpers/PerDistrict';
+import SectionHeading from '../components/SectionHeading';
+import EscapeListener from '../components/EscapeListener';
+import FilterBar from '../components/FilterBar';
+import SelectTimeRange, {
+  momentRange,
+  timeRangeText,
+  TIME_RANGE_45_DAYS_AGO
+} from '../components/SelectTimeRange';
+import {ALL} from '../components/SimpleFilterSelect';
+import SelectGrade from '../components/SelectGrade';
+import SelectHouse from '../components/SelectHouse';
+import SelectCounselor from '../components/SelectCounselor';
+import DashboardBarChart from '../school_administrator_dashboard/DashboardBarChart';
+import * as dashboardStyles from '../school_administrator_dashboard/dashboardStyles';
+import StudentsTable from './StudentsTable';
+import SelectExcusedAbsences, {
+  EXCLUDE_EXCUSED_ABSENCES,
+  ALL_ABSENCES
+} from './SelectExcusedAbsences';
+
+
+export default class SchoolAbsencesDashboard extends React.Component {
+  constructor(props, context) {
+    super(props);
+
+    this.state = initialState(props, context);
+    this.onResetFilters = this.onResetFilters.bind(this);
+    this.onExcusedAbsencesChanged = this.onExcusedAbsencesChanged.bind(this);
+    this.onGradeChanged = this.onGradeChanged.bind(this);
+    this.onHouseChanged = this.onHouseChanged.bind(this);
+    this.onCounselorChanged = this.onCounselorChanged.bind(this);
+    this.onTimeRangeKeyChanged = this.onTimeRangeKeyChanged.bind(this);
+    this.onClearHomeroomSelected = this.onClearHomeroomSelected.bind(this);
+    this.onBarChartColumnClicked = this.onBarChartColumnClicked.bind(this);
+    this.memoize = memoizer();
+  }
+
+  // Props changes won't happen as this component is currently written,
+  // so this is just defensive.
+  componentWillReceiveProps() {
+    this.memoize = memoizer();
+  }
+
+  shouldIncludeExcusedAbsences() {
+    const {excusedAbsencesKey} = this.state;
+    return excusedAbsencesKey === ALL_ABSENCES;
+  }
+
+  filteredStudents(options = {}) {
```

Comment:
@edavidsonsawyer hello! :)  This is the pull request I mentioned, where the goal is to be able to write functions like `filteredStudents` and `filteredAbsences` that just do simple ifs for whatever filters we want.

So here I wrote this naively, then looked in the Chrome profiler to see what was slowest, and then added the calls to `this.memoize` which memoizes different computations so they only have to be done once.  So even if the code I wrote calls `this.filteredStudents` lots of times it only has to actually compute it once.

Reply:
@kevinrobinson this is cool! It's a much cleaner approach. I also like switching out the monthly rate since as you mentioned that doesn't make a lot of sense over the time ranges users are seeing (and its faster!). 

I do wonder whether it would be useful to have some way of looking at attendance over time when it comes to the 90 day and school year ranges, though I'm not sure whether anyone was getting value out of that. But then, the whole visualization of this should probably change somewhere in the future since, like you said, the student list is where the action is at.

Reply:
@edavidsonsawyer Yeah good question about attendance over time!  I was wondering that too, and ended up deciding to not do that, and to minimize the attendance rate number here (keeping the single bar chart where you have to hover to see a number vs. seeing the number more prominently).  The intention there was to focus attention on individual students and immediate actions they can take, rather than understanding aggregate distributions or patterns or trends.  In other words, we're trying to help educators notice things happening with individual students in a way that's directly tied to actions they can take, rather than presenting them with an abstract metric that they'll be held responsible to drive up or down, as people might do in other kinds of settings.